pkg/router/template/template_helper.go

```diff
@@ -170,6 +171,71 @@ func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
 	return endpoints
 }
 
+// getHTTPAliasesGroupedByHost returns HTTP(S) aliases grouped by their host.
+func getHTTPAliasesGroupedByHost(aliases map[string]ServiceAliasConfig) map[string]map[string]ServiceAliasConfig {
+	result := make(map[string]map[string]ServiceAliasConfig)
+
+	for k, a := range aliases {
+		if a.TLSTermination == routeapi.TLSTerminationPassthrough {
+			continue
+		}
+
+		if _, exists := result[a.Host]; !exists {
+			result[a.Host] = make(map[string]ServiceAliasConfig)
+		}
+		result[a.Host][k] = a
+	}
+
+	return result
+}
+
+// getPassthroughAliases returns non-HTTP(S) (TLS Passthrough) aliases.
+func getPassthroughAliases(aliases map[string]ServiceAliasConfig) map[string]ServiceAliasConfig {
+	result := make(map[string]ServiceAliasConfig)
+
+	for k, a := range aliases {
+		if a.TLSTermination == routeapi.TLSTerminationPassthrough {
+			result[k] = a
+		}
+	}
+
+	return result
+}
+
+// getPrimaryAliasKey returns the key of the primary alias for a group of aliases.
+// It is assumed that all input aliases have the same host.
+// In case of a single alias, the primary alias is the alias itself.
+// In case of multiple alias with no TSL termination (Edge or Passthrough),
+// the primary alias is the alphabetically first alias.
+// In case of multiple aliases, some of them with TLS termination, the primary alias is
+// the alphabetically first alias among the TLS aliases.
+func getPrimaryAliasKey(aliases map[string]ServiceAliasConfig) string {
+	if len(aliases) == 0 {
+		return ""
+	}
+
+	if len(aliases) == 1 {
+		for k := range aliases {
+			return k
+		}
+	}
+
+	var keys []string
+	for k := range aliases {
+		keys = append(keys, k)
+	}
+
+	sort.Strings(keys)
```

Comment:
`sort.Sort(sort.Reverse(sort.StringSlice(keys)))`

Apologies, my bad on this one. I said alphabetically first in an email I had sent you. I just realized that the current sorting in the equivalent haproxy reload script is actually reversed (`sort -r`), so its the last one alphabetically.